tardis/io/atom_data/base.py

```diff
@@ -321,41 +290,57 @@ def __init__(
 
         self.atom_data = atom_data
         self.ionization_data = ionization_data
-        self.levels = levels
-        # Not sure why this is need - WEK 17 Sep 2023
+        self.levels = SimpleAtomData(levels).data
+        # Cast to float so that Numba can use the values in numpy functions
         self.levels.energy = self.levels.energy.astype(np.float64)
-        self.lines = lines
+        self.lines = SimpleAtomData(lines).data
+
+        collected_macro_atom_data = MacroAtomData(
+            macro_atom_data, macro_atom_references
+        )
 
         # Rename these (drop "_all") when `prepare_atom_data` is removed!
-        self.macro_atom_data_all = macro_atom_data
-        self.macro_atom_references_all = macro_atom_references
+        self.macro_atom_data_all = (
+            collected_macro_atom_data.transition_probability_data
+        )
+        self.macro_atom_references_all = (
+            collected_macro_atom_data.block_reference_data
+        )
+
+        self.zeta_data = SimpleAtomData(zeta_data).data
 
-        self.zeta_data = zeta_data
+        collected_collision_data = CollisionData(
+            collision_data, collision_data_temperatures, yg_data
+        )
 
-        self.collision_data = collision_data
-        self.collision_data_temperatures = collision_data_temperatures
+        self.collision_data = collected_collision_data.data
+        self.collision_data_temperatures = collected_collision_data.temperatures
 
-        self.synpp_refs = synpp_refs
+        self.synpp_refs = SimpleAtomData(synpp_refs).data
```

Comment:
I see a lot of `se.fx = SimpleAtomData(x).data` which from what I can see isn't doing anything except assigning `self.x=x`, why have this wrapper and the extra step here?

Reply:
See previous comments- this wrapper should theoretically provide a uniform interface for pre- and post- sliced data. But we can do that with pure dataframes if we prefer (though there are a lot of little weird things done to the dataframes in the current atomdata class)